contracts/Truebit.sol

```diff
@@ -1,57 +1,70 @@
 pragma solidity ^0.4.4;
-import "./Requester.sol";
+import "./TaskGiver.sol";
 
 contract Truebit{
   mapping (uint => Task) tasks;
   uint numTasks;
   uint gasPrice = 1000000; //1 million wei/instruction 
 
-  struct Task{
+  struct Task {
     bytes32 dataRoot;
-    address requester;
+    address taskGiver;
     uint    minDeposit;
     uint    gasLimit;
 
     bool solved;
-    bool proccessed;
+    bool processed;
     bytes32 globalRoot;
     bytes32 solution; //can be a root
   }
 
-  struct Solution{
+  function solved(Task t) private returns (bool){
+    return t.solved; //stub for now
+  }
+
+  function cheatSolve(uint taskIndex, bytes32 globalRoot, bytes32 solution){
+    Task t = tasks[taskIndex];
+    t.globalRoot = globalRoot;
+    t.solution = solution;
+    t.solved = true;
   }
 
+  event PostTask(
+    uint taskId,
+    uint minDeposit
+  );
+
+  event PostBid(
+    uint indexed taskId,
+    address from
+  );
+
   function createTask(bytes32 dataRoot, uint minDeposit) payable returns (uint){
     uint gasLimit = msg.value / gasPrice;
     tasks[numTasks] = Task(dataRoot, msg.sender, minDeposit, gasLimit, false, false, 0x0, 0x0);
+    PostTask(numTasks, minDeposit);
     numTasks++;
     return numTasks - 1;
   }
 
-  function proccessTask(uint taskIndex){
+  function processTask(uint taskIndex){
     Task t = tasks[taskIndex];
-    require(solved(t));
-    require(!t.proccessed);
```

Comment:
Why did you remove these two lines?

Reply:
Must've done that when I was messing around with testing. Added them back.